crawler/storage/elastic_storage.py

```diff
@@ -0,0 +1,120 @@
+# © Copyright 2021 HP Development Company, L.P.
+import os
+import zipfile
+import io
+from datetime import datetime
+from urllib.parse import urlparse
+from re import subn
+from elasticsearch import Elasticsearch, helpers
+
+from utils import SubCrawlColors, SubCrawlHelpers
+from .default_storage import DefaultStorage
+
+
+class ElasticStorage(DefaultStorage):
+
+    cfg = None
+    logger = None
+    es = None
+    index = None
+    archive_location = None
+    archive_content = False
+    max_fields = 0
+
+    def __init__(self, config, logger):
+        self.cfg = config
+        self.logger = logger
+        self.archive_location = SubCrawlHelpers.get_config(self.cfg,'elasticsearch', 'archive_log_location')
+        self.archive_content = SubCrawlHelpers.get_config(self.cfg, 'elasticsearch', 'archive_response_content')
+        self.index = SubCrawlHelpers.get_config(self.cfg, 'elasticsearch', 'index')
+
+        try:
+            self.es = Elasticsearch([{'host': SubCrawlHelpers.get_config(self.cfg, 'elasticsearch', 'host'),
+                                    'port': SubCrawlHelpers.get_config(self.cfg, 'elasticsearch', 'port'),
+                                    'index': self.index}])
+            self.es.ping()
+
+            if self.es.indices.exists(self.index) == False:
+                self.logger.debug('[ELASTIC] Index did not exist, creating: ' + self.index)
+                self.es.indices.create(index=self.index)
+
+            if self.archive_content:
+                if not os.path.isdir(self.archive_location):
+                    os.mkdir(self.archive_location)
+                    self.logger.debug('[ELASTIC] Response content being saved, log created at: ' + self.archive_location)
+            
+        except Exception as e:
+            self.logger.error('[ELASTIC] Problem connecting to Elastic: ' + str(e))
+            raise e
+
+    def load_scraped_domains(self):
+        return []
+
+    def normalize_field_name(self, field_name):
+        return field_name.replace(' ','-').replace('-','_').lower()
```

Comment:
Note that the second replace function replaces the already replaced elements of the first function again. In the end this would correspond to:
```
field_name.replace(' ','_').replace('-','_').lower()
```
If you want the spaces to remain minus signs, the order of the functions must be reversed.

Reply:
Good catch :) You are correct, all replacements should end up in an underscore.